### PR TITLE
enh(resources): use resource types filtering as 'providers'

### PR DESCRIPTION
**Pull request by adr-mo**
First PR of the integration of Anomaly Detection services into Resource Status.
This PR intends to use resources types as 'providers' to ease the integration of the type 'anomaly-detection'

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Fully tested afterwards

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: config/packages/validator/validation.yaml [hunk 1]
```diff
@@ -187,13 +187,10 @@ Centreon\Domain\Monitoring\ResourceFilter:
...
                 type: array
                 groups: [Default]
-            - Choice:
-                choices:
-                    - service
-                    - host
-                    - metaservice
-                multiple: true
-                groups: [Default]
+            - All:
```
Comment (adr-mo) [inline config/packages/validator/validation.yaml:190]: Check configuration

File: src/Centreon/Application/Controller/MonitoringResourceController.php [hunk 2]
```diff
@@ -218,6 +218,21 @@ public function list(
             'json'
         );
 
+        /**
```
Comment (adr-mo) [inline src/Centreon/Application/Controller/MonitoringResourceController.php]: Add private method to check filter types provided.

File: src/Centreon/Application/Controller/MonitoringResourceController.php [hunk 3]
```diff
@@ -128,34 +134,28 @@ class MonitoringResourceController extends AbstractController
...
-        UrlGeneratorInterface $router,
-        IconUrlNormalizer $iconUrlNormalizer
+        protected ResourceServiceInterface $resource,
+        protected UrlGeneratorInterface $router,
+        protected IconUrlNormalizer $iconUrlNormalizer,
+        \Traversable $resourceTypes
     ) {
         $this->resource = $resource;
         $this->router = $router;
         $this->iconUrlNormalizer = $iconUrlNormalizer;
```
Comment (adr-mo) [inline src/Centreon/Application/Controller/MonitoringResourceController.php]:
```suggestion
```

File: src/Centreon/Infrastructure/Monitoring/Resource/DbReadResourceRepository.php [hunk 4]
```diff
@@ -125,7 +139,7 @@ public function setContact(ContactInterface $contact): ResourceRepositoryInterfa
      */
     public function filterByAccessGroups(?array $accessGroups): ResourceRepositoryInterface
     {
-        $this->accessGroups = $accessGroups;
+        $this->accessGroups = $accessGroups === null ? [] : $accessGroups;
```
Comment (adr-mo) [inline src/Centreon/Infrastructure/Monitoring/Resource/DbReadResourceRepository.php]:
```suggestion
        $this->accessGroups = $accessGroups ?? [];
```

File: src/Core/Domain/RealTime/Model/ResourceTypes/ServiceResourceType.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+{
+    /**
+     * @var string $type
+     */
+    protected string $type = 'service';
+
+    /**
+     * @var integer $id
+     */
+    protected int $id = 0;
```
Comment (adr-mo) [inline src/Core/Domain/RealTime/Model/ResourceTypes/ServiceResourceType.php]: Use constant for all resource types for id and type

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -16916,3 +16916,6 @@ msgstr "Une erreur s'est produite lors de la récupération des catégories d'h
 
 msgid "Warning, maximum size exceeded for input '%s' (max: %d), it will be truncated upon saving"
 msgstr "Attention, taille maximale dépassée pour le champ '%s' (max: %d), il sera tronqué à l'enregistrement"
+
+msgid "Unsupported resource type provided"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgid "Resource type provided not supported"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16916,3 +16916,6 @@ msgstr "Une erreur s'est produite lors de la récupération des catégories d'h
...
+msgstr "Type de resource fourni non supporté"
```
Comment (smau-centreon) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po]:
```suggestion
msgstr "Type de ressource fourni non supporté"
```

**Review by smau-centreon — CHANGES_REQUESTED** [review]

**Review by a-launois — APPROVED** [review]

**Review by smau-centreon — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**